File: airbyte-config/config-models/src/test/java/io/airbyte/config/storage/MinioS3ClientFactoryTest.java
/*
 * Copyright (c) 2020-2024 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.config.storage;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;

import org.junit.jupiter.api.Test;

class MinioS3ClientFactoryTest {

  @Test
  void testMinio() {
    final var bucket = new StorageBucketConfig("log", "state", "workload");
    final var config = new MinioStorageConfig(bucket, "access", "secret", "http://endpoint.test");

    assertDoesNotThrow(() -> new MinioS3ClientFactory(config).get());
  }

}

File: airbyte-config/config-models/src/test/kotlin/io/airbyte/config/storage/MinioS3ClientFactoryTest.kt
package io.airbyte.config.storage

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertDoesNotThrow

val buckets = StorageBucketConfig(log = "log", state = "state", workloadOutput = "workload", activityPayload = "payload")
val config = MinioStorageConfig(buckets = buckets, accessKey = "access", secretAccessKey = "secret", endpoint = "http://endpoint.test")

class MinioS3ClientFactoryTest {
  @Test
  fun `minio doesn't throw exception`() {
    assertDoesNotThrow { MinioS3ClientFactory(config).get() }
  }
}
